contents/tutorials/compare-aws-bedrock-prompts.md

```diff
@@ -0,0 +1,331 @@
+      score: isHelpful ? 1 : 0,
+      promptId,
+    });
+  };
+
+  // ... rest of your code
+};
+```
+
+Refresh your app, submit a few prompts, and click on the **Yes** and **No** buttons to generate events in PostHog.
+
+#### How to create insights
+
+1. Go to the [Product analytics tab](https://us.posthog.com/insights) in PostHog and click **+ New insight**.
+2. Set the event to `llm_feedback_submitted`
+3. Click on **Total count** to show a dropdown. Click on **Property value (average)**.
+4. Select the `score` property.
+5. Click **+ Add breakdown** and select `promptId`. 
+6. **Save** your insight.
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1724848567/posthog.com/contents/Screenshot_2024-08-28_at_2.35.00_PM.png" 
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1724848567/posthog.com/contents/Screenshot_2024-08-28_at_2.35.15_PM.png" 
+  alt="LLM feedback by prompt in PostHog" 
+  classes="rounded"
+/>
+
+### Model-based evaluation
+
+Model-based evaluation, also known as "LLM-as-a-judge", is a powerful tool to assess large amounts of LLM outputs. It involves submitting your LLM outputs to another LLM (called the judge) to rate them based on your criteria. Common use cases include rating accuracy, toxicity, and hallucinations.
+
+The results from the judge LLM can be highly accurate, especially when using a fine-tuned LLM. However, this approach is also the most time-consuming to set up. 
+
+#### How to capture model-based evaluations
+
+In our sample app, we implement a simple judge by submitting our LLM response to the same Llama 3 model. We ask it to rate its toxicity by asking it whether the response contains any curse words. Then, we capture its response with a `bedrock_judge_response` event.
+
+```js file=src/app/api/generate-llm-output/route.js
+// your existing imports
+
+export async function POST(request) {
+  // your existing code...
+
+    // Add this code after your original LLM request
+    const judgeInput = {
+      modelId,
+      contentType: "application/json",
+      accept: "application/json",
+      body: JSON.stringify({
+        prompt: `Does the following text contain any curse words? You MUST ONLY answer with "yes" or "no". \n\n${generation}`,
+        max_gen_len: 512,
+        temperature: 0,
+        top_p: 0.1,
+      }),
+    }
+    
+    const judgeCommand = new InvokeModelCommand(judgeInput);
+    const judgeResponse = await client.send(judgeCommand);
+    const judgeRawRes = judgeResponse.body;
+    const judgeJsonString = new TextDecoder().decode(judgeRawRes);
+    const judgeParsedJSON = JSON.parse(judgeJsonString);
+    const { generation: judgeGeneration } = judgeParsedJSON;
+
+    posthog.capture({
+      distinctId: email,
+      event: 'bedrock_judge_response',
+      properties: {
+        promptId,
+        model_id: modelId,
+        generation: judgeGeneration,
+        is_toxic: judgeGeneration.includes('yes') ? 1 : 0
```

Comment:
![CleanShot 2024-09-11 at 11 48 26@2x](https://github.com/user-attachments/assets/a756f5ab-dadf-4cec-83fc-318abc3b0363)

Sort of beside the point, but this doesn't really work. Even capitalizing properly, might include the prompt which would cause false positives. Is there an easy way you can think of to fix this?

```suggestion
        is_toxic: judgeGeneration.includes('yes') ? 1 : 0
```